kuma/users/views.py

```diff
@@ -450,14 +455,87 @@ def user_edit(request, username):
 
 
 @redirect_in_maintenance_mode
+@login_required
 def user_delete(request, username):
+    user = get_object_or_404(User, username=username)
+    if user != request.user:
+        return HttpResponseForbidden()
 
-    edit_user = get_object_or_404(User, username=username)
+    def donate_attributions():
+        anon, _ = User.objects.get_or_create(username='Anonymous')
+        user.created_revisions.update(creator=anon)
+        user.created_attachment_revisions.update(creator=anon)
+
+    def scrub_user():
+        # From the User abstract class
+        user.first_name = ''
+        user.last_name = ''
+        user.email = ''
+
+        # All User attributes
+        user.timezone = ''
+        user.locale = ''
+        user.homepage = ''
+        user.title = ''
+        user.fullname = ''
+        user.organization = ''
+        user.location = ''
+        user.bio = ''
+        user.irc_nickname = ''
+        user.website_url = ''
+        user.github_url = ''
+        user.mozillians_url = ''
+        user.twitter_url = ''
+        user.linkedin_url = ''
+        user.facebook_url = ''
+        user.stackoverflow_url = ''
+        user.discourse_url = ''
+        user.stripe_customer_id = ''
+        user.save()
+
+        user.socialaccount_set.all().delete()
+
+    def force_logout():
+        request.session.clear()
+
+    def delete_user():
+        user.revisionakismetsubmission_set.all().delete()
```

Comment:
Why keep them? The `AkismetSubmission` is ["used as a log of sorts"](https://github.com/mdn/kuma/blob/3ea95bb7d3c53a6d1a08eaf45b5f6b66b80c0e39/kuma/spam/models.py#L22) and if we replace the user with `anon`, what point is it to keep. All it will tell you is that 'ham' or 'spam' was submitted at {date} by some anonymous user. 

I'm just not convinced there's any value in keep the data. Granted, if it is a log, we also don't open the /var/log/someapp/things.log file and rewrite it.

Reply:
I'm thinking it's like removing an entry from a log when there's still value in the entry. We have the information that an Akismet "ham" or "spam" submission was made at a specific time/date on a specific document revision, and that information still seems valuable even though the sender is now anonymous.

Reply:
Nevermind, I changed so that they're moved over to the anonymous user.